### PR TITLE
Support conditional testing for NM provider

### PR DESCRIPTION
**Pull request by liangwen12year**
Current Network role can not ensure test playbook running with both
initscripts provider and conditional NM provider(the test running with NM
provider which demanding for NM mininum version or extra running
condition). This PR is intended to fix the bug.

### CODE REVIEW

**Comment by tyll** [comment]
:+1:

File: .yamllint.yml [hunk 1]
```diff
@@ -21,3 +21,4 @@ rules:
     ignore: |
```
Comment (tyll) [inline .yamllint.yml:21]: Regarding the commit message:

```

Ensure testing with both initscripts and conditional NM provider

```

Proposal: Support conditional testing for NM provider

```
Current Network role can not ensure test playbook running with both
initscripts provider and conditional NM provider(the test running with NM
provider which demanding for NM mininum version or extra running
condition). This PR is intended to fix the bug.

In the mean time, enable the support for ethtool coalesce option with
initscripts provider.
```

Proposal:

```
- To properly run tests for features that are supported both in initscripts and NM but require a specific NM version or other condition, add a new test category for these tests. 
- Use the new test category to enable testing coalesce support with initscripts
```

Comment (liangwen12year) [reply .yamllint.yml:21]: Commit message changed, thanks!

**Comment by liangwen12year** [comment]
CI failed caused by 
```
TASK [Get profile's coalescing options] **************************************************************************************************************************************************
task path: /home/wenliang/network/tests/playbooks/tests_ethtool_coalesce.yml:41
fatal: [CentOS-8-GenericCloud-8.3.2011-20201204.2.x86_64.qcow2]: FAILED! => {"changed": true, "cmd": ["nmcli", "-g", "ethtool.coalesce-rx-frames", "c", "show", "testnic1"], "delta": "0:00:00.013299", "end": "2021-04-27 19:21:17.341112", "msg": "non-zero return code", "rc": 10, "start": "2021-04-27 19:21:17.327813", "stderr": "Error: testnic1 - no such connection profile.", "stderr_lines": ["Error: testnic1 - no such connection profile."], "stdout": "", "stdout_lines": []}
```

So `./tests/playbooks/tests_ethtool_coalesce.yml` need to be adjusted.

**Comment by liangwen12year** [comment]
> CI failed caused by
> 
> ```
> TASK [Get profile's coalescing options] **************************************************************************************************************************************************
> task path: /home/wenliang/network/tests/playbooks/tests_ethtool_coalesce.yml:41
> fatal: [CentOS-8-GenericCloud-8.3.2011-20201204.2.x86_64.qcow2]: FAILED! => {"changed": true, "cmd": ["nmcli", "-g", "ethtool.coalesce-rx-frames", "c", "show", "testnic1"], "delta": "0:00:00.013299", "end": "2021-04-27 19:21:17.341112", "msg": "non-zero return code", "rc": 10, "start": "2021-04-27 19:21:17.327813", "stderr": "Error: testnic1 - no such connection profile.", "stderr_lines": ["Error: testnic1 - no such connection profile."], "stdout": "", "stdout_lines": []}
> ```
> 
> So `./tests/playbooks/tests_ethtool_coalesce.yml` need to be adjusted.

It seems that we also can not get veth ring setting and coalesce setting: 
```
[root@ibm-p8-kvm-03-guest-02 ~]# ethtool -g testnic1
Ring parameters for testnic1:
Cannot get device ring settings: Operation not supported
[root@ibm-p8-kvm-03-guest-02 ~]# ethtool -c testnic1
Coalesce parameters for testnic1:
Cannot get device coalesce settings: Operation not supported
```

**Comment by tyll** [comment]
The `ifup` deprecation is expected, there is nothing to do for the network role at this point.

I did not look into the problem regarding the options, maybe veth interfaces don't support this and we need a different solution.

**Comment by cathay4t** [comment]
@liangwen12year Both tox and integration test is failing. Please take a look. The code looks good to me through.

**Comment by richm** [comment]
[citest bad]

**Comment by richm** [comment]
[citest pending]

**Comment by liangwen12year** [comment]
@richm , `tox -e molecule` also failed on the main branch
```
Traceback (most recent call last):
  File "/home/wenliang/network/.tox/molecule/bin/yamllint", line 8, in <module>
    sys.exit(run())
  File "/home/wenliang/network/.tox/molecule/lib/python3.8/site-packages/yamllint/cli.py", line 188, in run
    conf = YamlLintConfig(file='.yamllint.yml')
  File "/home/wenliang/network/.tox/molecule/lib/python3.8/site-packages/yamllint/config.py", line 44, in __init__
    self.parse(content)
  File "/home/wenliang/network/.tox/molecule/lib/python3.8/site-packages/yamllint/config.py", line 94, in parse
    base = YamlLintConfig(file=path)
  File "/home/wenliang/network/.tox/molecule/lib/python3.8/site-packages/yamllint/config.py", line 41, in __init__
    with open(file) as f:
FileNotFoundError: [Errno 2] No such file or directory: 'yamllint_defaults.yml'
ERROR: InvocationError for command /usr/bin/bash /home/wenliang/.local/lib/python3.8/site-packages/tox_lsr/test_scripts/runsyspycmd.sh molecule lint -s default (exited with code 1)
____________________________________________________________________________________ summary ____________________________________________________________________________________
ERROR:   molecule: commands failed
```

**Comment by richm** [comment]
> @richm , tox -e molecule also failed on the main branch

Make sure you are using the latest tox-lsr `pip install --upgrade --user "git+https://github.com/linux-system-roles/tox-lsr@2.4.0"`

Then, do a tox refresh `-r` - `tox -r -e molecule`

**Comment by richm** [comment]
> @richm , tox -e molecule also failed on the main branch

Unfortunately because of the `black` error, the github actions tox CI is being terminated before it gets to the molecule testing in `tox / python (3.8)` so I can't see what is the problem there.

**Comment by liangwen12year** [comment]
> > @richm , tox -e molecule also failed on the main branch
> 
> Unfortunately because of the `black` error, the github actions tox CI is being terminated before it gets to the molecule testing in `tox / python (3.8)` so I can't see what is the problem there.

Do we need to fix those `black` error in a separate PR ? If needed, I can do it.

**Comment by richm** [comment]
> > > @richm , tox -e molecule also failed on the main branch
> > 
> > 
> > Unfortunately because of the `black` error, the github actions tox CI is being terminated before it gets to the molecule testing in `tox / python (3.8)` so I can't see what is the problem there.
> 
> Do we need to fix those `black` error in a separate PR ? If needed, I can do it.

https://github.com/linux-system-roles/network/pull/389

**Comment by jharuda** [comment]
[citest pending]

**Comment by richm** [comment]
@liangwen12year you'll need to rebase and repush your branch to pick up the python black changes. I don't anticipate any rebase conflicts, but please check.

**Comment by liangwen12year** [comment]
> @liangwen12year you'll need to rebase and repush your branch to pick up the python black changes. I don't anticipate any rebase conflicts, but please check.

Rebased and repushed.

**Comment by richm** [comment]
[citest pending]

**Comment by richm** [comment]
[citest bad]

**Comment by richm** [comment]
[citest pending]

**Review by ffmancera — APPROVED** [review]
LTGM, thanks! Great patch! Please, rebase it.

**Comment by liangwen12year** [comment]
> LTGM, thanks! Great patch! Please, rebase it.

Conflict resolved and rebased. Thanks.

**Pull request merged**